cpp/oneapi/dal/backend/primitives/test/ndarray.cpp

```diff
@@ -489,6 +506,47 @@ TEST_M(ndarray_test, "can assign ndarray", "[ndarray]") {
     sycl::free(x_ptr, queue);
 }
 
+TEST_M(ndarray_test, "can be flattened with host usm", "[ndarray]") {
+    DECLARE_TEST_POLICY(policy);
+    auto& queue = policy.get_queue();
+
+    auto [x, event] = ndarray<float, 2>::ones(queue, { 7, 5 }, sycl::usm::alloc::host);
+
+    event.wait_and_throw();
+
+    auto raw_arr = x.flatten();
+
+    for (std::int64_t i = 0; i < x.get_count(); ++i) {
+        REQUIRE(raw_arr[i] == 1);
+    }
+}
+
+TEST_M(ndarray_test, "can be flattened with device usm", "[ndarray]") {
+    DECLARE_TEST_POLICY(policy);
+    auto& queue = policy.get_queue();
+
+    const std::int64_t m = 7;
+    const std::int64_t n = 5;
+
+    auto [x, event] = ndarray<float, 2>::ones(queue, { m, n }, sycl::usm::alloc::device);
+
+    event.wait_and_throw();
+
+    const auto raw_arr = x.flatten();
+
+    const float* raw_ptr = raw_arr.get_data();
+
+    float* res_ptr = sycl::malloc_shared<float>(n * m, queue);
+
+    queue
+        .submit([&](sycl::handler& h) {
+            h.memcpy(res_ptr, raw_ptr, (n * m));
+        })
+        .wait_and_throw();
+
+    sycl::free(res_ptr, queue);
```

Comment:
I suggest to use our wrappers instead raw memory allocation.

```c++
REQUIRE(raw_arr.get_count() == x.get_count());
const auto raw_arr_host = te::matrix<float>::wrap(queue, raw_arr).to_host();
const float*raw_arr_host_ptr = raw_arr_host.get_data();
for (std::int64_t i = 0; i < x.get_count(); ++i) {
    REQUIRE(raw_arr_host_ptr[i] == 1.0f);
}
```

Reply:
I suppose that it this test should be simple and not require additional entities.

Reply:
Fixed